bindings_generator/src/lib.rs

```diff
@@ -177,14 +152,89 @@ fn generate_class_bindings(
     Ok(())
 }
 
-fn rust_safe_name(name: &str) -> &str {
+fn rust_safe_name(name: &str) -> proc_macro2::Ident {
     match name {
-        "use" => "_use",
-        "type" => "_type",
-        "loop" => "_loop",
-        "in" => "_in",
-        "override" => "_override",
-        "where" => "_where",
-        name => name,
+        "use" => format_ident!("_use"),
+        "type" => format_ident!("_type"),
+        "loop" => format_ident!("_loop"),
+        "in" => format_ident!("_in"),
+        "override" => format_ident!("_override"),
+        "where" => format_ident!("_where"),
+        name => format_ident!("{}", name),
+    }
+}
+
+#[cfg(feature = "debug")]
+#[cfg(test)]
```

Comment:
Because the generated tokens are written to a stream and are not being used in a macro, we don't get that check. This really helped find chunks of code that rustfmt couldn't parse and dump it to a file.
It'd be great if we could use this work as a stepping stone towards what you described.

Reply:
I see. Could you use something like [`tempdir`](https://crates.io/crates/tempfile) for the `bad_code.rs` file, though? I don't think writing to the current directory is a very good thing to do here.

Reply:
I thought about using `tempfile` but didn't because we didn't want the temporary file to be deleted. It's only created when there is 'bad code'. Thoughts? I'm hoping that this can go away by turning this code generation into macros that are then used by `gdnative-core` and `gdnative-binding`. Doing that conversion should give what you first mentioned, proper compiler errors.

Reply:
Maybe just create it in `std::env::temp_dir` for now?